operator/e2e/e2e_test.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/rand"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var _namespace = "pulumi-kubernetes-operator"
+
+func TestE2E(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	// Generate a random tag to ensure pods get re-created when re-running the
+	// test locally.
+	bytes := make([]byte, 12)
+	_, _ = rand.Read(bytes) //nolint:staticcheck // Don't need crypto here.
+	tag := hex.EncodeToString(bytes)
+	projectimage := "pulumi/pulumi-kubernetes-operator-v2:" + tag
```

Comment:
Using a `latest` tag with an `IfNotPresent` pull policy is sufficient for CI. I was having trouble running things multiple times locally because a new latest tag wouldn't cause things to re-deploy but I can probably simplify this with some more robust teardown logic.